src/saveload/afterload.cpp

```diff
@@ -3038,21 +3039,19 @@ bool AfterLoadGame()
 	if (IsSavegameVersionBefore(SLV_EXTEND_INDUSTRY_CARGO_SLOTS)) {
 		/* Make sure added industry cargo slots are cleared */
 		for (Industry *i : Industry::Iterate()) {
-			for (auto it = std::begin(i->produced) + 2; it != std::end(i->produced); ++it) {
-				it->cargo = INVALID_CARGO;
-				it->waiting = 0;
-				it->rate = 0;
-				it->history = {};
-			}
-			for (auto it = std::begin(i->accepted) + 3; it != std::end(i->accepted); ++it) {
-				it->cargo = INVALID_CARGO;
-				it->waiting = 0;
-			}
+			/* Remove all entries from the first with invalid cargo */
```

Comment:
This comment (repeated several times) seems to be missing a word for me to understand it. Remove all entries from the first...what? First slot? First industry? 

(The code is beyond me or I'd answer my own question.) 🙂

Reply:
From the first slot onwards. But looking at this, along with the recent cargo type changes, I think the code needs to be revised anyway.